tests/python/pants_test/rules/test_fmt_integration.py

```diff
@@ -0,0 +1,16 @@
+# Copyright 2019 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from pants_test.pants_run_integration_test import PantsRunIntegrationTest
+
+
+class FmtIntegrationTest(PantsRunIntegrationTest):
+  def test_black_no_python_sources_should_noop(self):
```

Comment:
I'd generify this name given that it's in the core `rules` folder. Something like `test_noop_when_no_fmt_available_for_target_type`